### PR TITLE
feature(ui-ux): pool swap - slippage tolerance design update

### LINKED ISSUE #2189: `pool swap` - slippage tolerance design update
#### What would you like to be added:
We would like to redesign the slippage tolerance to display the buttons (with custom option).
image

- [x] Users will be able to "click" easily on a slippage tolerance
- [x] Allow them to input custom slippage
- [x] Remove `20%` option and add `0.5%`

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #2189

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

**Comment by thedoublejay** [comment]
As discussed, let's remove the 20% by default and add 0.5%.

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx [hunk 1]
```diff
@@ -1,124 +1,211 @@
...
-            setIsSelectorOpen(true)
-            setSlippageToleranceToBottomSheet()
-            onPress()
-          }}
-          style={tailwind('p-4 pr-2')}
-          testID='slippage_select'
+      <View style={tailwind('px-4 mt-5 w-full')}>
+        <View
+          testID='slippage_tolerance_heading'
+          style={tailwind(['flex-row mb-3 items-center justify-between'])}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx:39]: Let's move the title closer to the selector
```suggestion
          style={tailwind(['flex-row items-center justify-between'])}
```

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx [hunk 2]
```diff
@@ -1,124 +1,211 @@
...
-      </View>
-      <View
-        style={tailwind('flex w-6/12 flex-row justify-end flex-wrap items-center')}
+      )}
+    </>
+  )
+}
+
+function SlippageButton ({ onPress, isActive, label, icon }: { onPress: () => void, isActive: boolean, label: string, icon?: ReactElement }): JSX.Element {
+  const buttonStyles = 'flex flex-row px-2 py-1.5 border border-gray-300 rounded items-center'
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx:190]: Can we add a different border color on dark mode? 

On light mode, the button border is consistent with other input field, but on dark mode, it is taking more attention than the other

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx [hunk 3]
```diff
@@ -1,124 +1,211 @@
...
+          light={tailwind('bg-transparent')}
+          style={tailwind('flex-row w-full')}
+        >
+          <WalletTextInput
+            autoFocus
+            onChangeText={onSlippageChange}
+            keyboardType='numeric'
+            autoCapitalize='none'
+            placeholder='0.00%'
+            style={tailwind('flex-grow h-8')}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx:162]: Let's add a width here so it doesn't overflow
IMG_FA0F798D9CEB-1

**Review by kyleleow — CHANGES_REQUESTED** [review]
Notice one focus/cursor shifting issue with custom slippage

Steps to reproduce:
- DEX screen 
- Tap on `Swap` for any pool pair
- Set any amount to swap
- Tap on `Custom` slippage
- Enter any percentage value
- Go back
- Tap on "Swap" on any pool pair again
- Enter any value to to swap
- Notice the cursor/focus move immediately to the custom slippage text input after entering one character in the swap amount input

**Comment by fullstackninja864** [comment]
@kyleleow resolved  `focus/cursor shifting issue with custom slippage`

**Pull request merged**